### PR TITLE
various: Add/update my copyright.

### PR DESCRIPTION
**Pull request by pfalcon**
Based on the git history, for modules I initially created or made substantial contributions to.

### CODE REVIEW

**Comment by pfalcon** [comment]
This PR also includes https://github.com/micropython/micropython/pull/4363 , which got ignored for more than a half-year now.

File: ports/unix/modos.c [hunk 1]
```diff
@@ -3,8 +3,8 @@
  *
  * The MIT License (MIT)
  *
+ * Copyright (c) 2014-2018 Paul Sokolovsky
  * Copyright (c) 2013, 2014 Damien P. George
- * Copyright (c) 2014 Paul Sokolovsky
```
Comment (dpgeorge) [inline ports/unix/modos.c]: Why is the ordered changed here (and elsewhere)?  Since 2013 is earlier than 2014 it makes sense to retain the order.

**Comment by dpgeorge** [comment]
Is it intended to update axtls?

File: ports/unix/modos.c (hunk 1 above)
Comment (pfalcon) [reply ports/unix/modos.c]: I changed order in files which were initially created by me. I hope you would agree that it makes sense if the original author were listed first.

Also, arguably, your copyright year would be adjusted to match the real history of a file, and if you remember, I did that once in a previous patchset on this matter, but refrained from doing this again, as ultimately, you're the best party to maintain your copyright notice.

(Finally, many copyright notices in files contributed by me in early period get your copyright mostly by cut-n-paste, with rationalization that while I create some module from scratch, the structure of it is governed (and copied) from your substantial effort. While later it became clear that it doesn't scale to have any MicroPython-related file ever written to include your copyright, I obviously don't try to revision notices added at that early time.)

**Comment by pfalcon** [comment]
> Is it intended to update axtls?

Definitely not, rebased.

File: ports/unix/modos.c (hunk 1 above)
Comment (dpgeorge) [reply ports/unix/modos.c]: Ok, you're right that this file looks to be completely fresh (or at least fresh enough) at its time of creation.  So I should adjust my year to match the history.

File: ports/unix/modtime.c [hunk 2]
```diff
@@ -3,6 +3,7 @@
  *
  * The MIT License (MIT)
  *
+ * Copyright (c) 2014-2017 Paul Sokolovsky
```
Comment (dpgeorge) [inline ports/unix/modtime.c]: Ok

File: ports/unix/modusocket.c [hunk 3]
```diff
@@ -3,8 +3,8 @@
  *
  * The MIT License (MIT)
  *
+ * Copyright (c) 2014-2018 Paul Sokolovsky
```
Comment (dpgeorge) [inline ports/unix/modusocket.c]: Ok

File: py/binary.c [hunk 4]
```diff
@@ -3,6 +3,7 @@
  *
  * The MIT License (MIT)
  *
+ * Copyright (c) 2014-2017 Paul Sokolovsky
```
Comment (dpgeorge) [inline py/binary.c]: Ok

File: py/binary.h [hunk 5]
```diff
@@ -3,6 +3,7 @@
  *
  * The MIT License (MIT)
  *
+ * Copyright (c) 2014 Paul Sokolovsky
```
Comment (dpgeorge) [inline py/binary.h]: Ok

File: py/frozenmod.h [hunk 6]
```diff
@@ -3,6 +3,7 @@
  *
  * The MIT License (MIT)
  *
+ * Copyright (c) 2015 Paul Sokolovsky
```
Comment (dpgeorge) [inline py/frozenmod.h]: Ok

**Comment by dpgeorge** [comment]
I merged the main commit here in 016d9a40fe822a624449ad10486706991a97be29, and a follow up in 653e1756c095a761dfea5bd91b94b9b2a150702c

For the change to LICENSE, that was not merged and can be discussed further in #4363

**Pull request closed without merging**